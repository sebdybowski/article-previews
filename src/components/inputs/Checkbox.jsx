import React from 'react';
import PropTypes from 'prop-types';

export const Checkbox = ({ label, value, onValueChange }) => <div className="form-check">
    <input className="form-check-input" type="checkbox" id={label} onChange={() => onValueChange(label)} checked={value} />
    <label className="form-check-label" htmlFor={label}>
        <span className="btn-link">{label}</span>
    </label>
</div>;

Checkbox.propTypes = {
    label: PropTypes.string.isRequired,
    value: PropTypes.bool,
    onValueChange: PropTypes.func.isRequired,
};