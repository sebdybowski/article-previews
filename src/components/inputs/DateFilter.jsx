import React from 'react';
import PropTypes from 'prop-types';

export const DateFilter = ({ onValueChange, sortOrder }) =>
    <button type="button" className="btn btn-link float-right" onClick={() => onValueChange()}>
        <span className={`sort-order ${sortOrder}`}>Sort by date</span>
    </button>;

DateFilter.propTypes = {
    onValueChange: PropTypes.func.isRequired,
    sortOrder: PropTypes.string,
};