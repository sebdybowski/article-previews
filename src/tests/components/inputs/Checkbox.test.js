import React from 'react';
import { shallow, render } from 'enzyme';
import { Checkbox } from "../../../components/inputs/Checkbox";
import _ from 'lodash';

const mockProps = {
    label: '',
    onValueChange: _.noop,
};

describe('Checkbox', () => {
    it('should render correctly', () => {
        const component = shallow(<Checkbox {...mockProps} />);
        expect(component).toMatchSnapshot();
    });
    it('should render correctly with true value', () => {
        const component = shallow(<Checkbox {...mockProps} value={true}/>);
        expect(component).toMatchSnapshot();
    });
    it('should render correctly with false value', () => {
        const component = shallow(<Checkbox {...mockProps} value={false}/>);
        expect(component).toMatchSnapshot();
    });
});