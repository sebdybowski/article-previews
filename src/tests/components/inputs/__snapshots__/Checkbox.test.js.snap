// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render correctly 1`] = `
<div
  className="form-check"
>
  <input
    className="form-check-input"
    id=""
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="form-check-label"
    htmlFor=""
  >
    <span
      className="btn-link"
    />
  </label>
</div>
`;

exports[`Checkbox should render correctly with false value 1`] = `
<div
  className="form-check"
>
  <input
    checked={false}
    className="form-check-input"
    id=""
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="form-check-label"
    htmlFor=""
  >
    <span
      className="btn-link"
    />
  </label>
</div>
`;

exports[`Checkbox should render correctly with true value 1`] = `
<div
  className="form-check"
>
  <input
    checked={true}
    className="form-check-input"
    id=""
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="form-check-label"
    htmlFor=""
  >
    <span
      className="btn-link"
    />
  </label>
</div>
`;
